Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst

```diff
@@ -0,0 +1,191 @@
+best practice not to rely on internal classes.
+
+You can find the complete class in the extension EXT:examples on GitHub:
+`GitHubLinkHandler <https://github.com/TYPO3-Documentation/t3docs-examples/blob/main/Classes/LinkHandler/GitHubLinkHandler.php>`__.
+
+We will explain some of the important methods bellow:
+
+Initialization and dependencies
+-------------------------------
+
+..  include:: /ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt
+
+For technical reasons not all dependencies needed by the backend link handler can
+be acquired by :ref:`DependencyInjection`. Therefore the following two methods
+are called by Core classes once the dependencies are available:
+
+:php:`LinkHandlerInterface::initialize()` takes care of setting the
+:php:`\TYPO3\CMS\Backend\Controller\AbstractLinkBrowserController`, the identifier and
+the configuration information. In this example we only need the configuration,
+the other parameters might be needed in different scenarios. The
+:php:`\TYPO3\CMS\Core\Page\PageRenderer` has also to be created here for as it
+is not possible to inject it at the point of creation of the backend link handler.
+
+The method :php:`setView()` is called by the :php:`AbstractLinkBrowserController`
+once the view is available and contains the necessary information to render
+the link browser window.
+
+..  note::
+    :php:`setView()` is not part of the :php:`LinkHandlerInterface`
+    and it being called is an implementation detail that might be subject to be
+    changed in the future.
+
+Render the link browser tab
+---------------------------
+
+The method :php:`LinkHandlerInterface::render()` is called when the tab should
+be rendered. It registers the required JavaScript in the page renderer, assigns
+variables to the view and returns the rendered HTML.
+
+..  include:: /ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerRender.rst.txt
+
+Set the link via JavaScript
+---------------------------
+
+When the button in the rendered form is clicked to set a link a custom
+JavaScript class interprets the form data and creates the link to be stored:
+
+..  todo: Configure code snippet tool to remove or shorten the license comment here
+
+..  include:: /ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_CustomLinkHandlerJavaScript.rst.txt
+
+It is important that the JavaScript function calls
+:js:`LinkBrowser.finalizeFunction()`. Otherwise no link will be set.
+
+If not done yet the JavaScript has to be registered in the file
+:file:`EXT:my_extension/Configuration/JavaScriptModules.php`. Otherwise it
+will not be found by :php:`$pageRenderer->loadJavaScriptModule()`.
+
+..  literalinclude:: _CustomLinkBrowser/_JavaScriptModules.php
+    :caption: EXT:examples/Configuration/JavaScriptModules.php
+
+As our JavaScript class depends on classes provided by EXT:backend,
+:php:`backend` has to be included in the dependencies. See also
```

Comment:
```suggestion
:php:`backend` has to be added as dependency. See also
```

More, what has to be done instead of where. The where is obvious from the snippet.